src/modal/components/ItemModal.jsx

```diff
@@ -40,6 +40,16 @@ const ItemModal = ({ item = {}, onClose, onSave, onDelete }) => {
       }}
       secondaryText={t("cancel")}
       onSecondaryClick={onClose}
+      thirdText={t("add_another_item")}
+      onThirdClick={() => {
+        if (!state.name || state.amount === undefined){
+          return;
+        }
+        onSave(state);
```

Comment:
Also, in react you shouldn't mutate the state. You should create a new object and use the setter to clupdate the state